src/rsvc/srv.c

```diff
@@ -656,18 +686,31 @@ stop(struct ds_rsvc *svc, bool destroy)
 int
 ds_rsvc_stop(enum ds_rsvc_class_id class, daos_iov_t *id, bool destroy)
 {
-	struct ds_rsvc *svc;
-	d_list_t       *entry;
+	struct ds_rsvc		*svc;
+	struct ds_rsvc_class	*impl;
+	d_list_t		*entry;
+	int			 rc;
 
+	impl = rsvc_class(class);
 	entry = d_hash_rec_find(&rsvc_hash, id->iov_buf, id->iov_len);
 	if (entry == NULL) {
-		return -DER_ALREADY;
+		rc = -DER_ALREADY;
+		goto out;
 	}
 	svc = rsvc_obj(entry);
 
 	d_hash_rec_delete_at(&rsvc_hash, &svc->s_entry);
 
-	return stop(svc, destroy);
+	rc = stop(svc, destroy);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to stop %s service: %d\n",
+			DP_UUID(id->iov_buf), impl->sc_classname, rc);
```

Comment:
id->iov_buf may not be an actual UUID; one must convert it into a name string using sc_name. We can drop the D_ERROR here and log in callers' code.